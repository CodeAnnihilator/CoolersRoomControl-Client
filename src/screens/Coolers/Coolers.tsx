import React, {ReactText, Component} from 'react';
import {ScrollView} from 'react-native';
import {NavigationComponent} from 'react-navigation';

import CoolerItem from './components/CoolerItem/CoolerItem';

import styles from './coolersStyles';

interface ICooler {
	id: number;
	power: string;
	title: string;
	series: string;
	status: string;
	room: string;
	place: string;
	source: string;
	type: string;
	certification: string;
	cop: string;
}

interface IProps {
	coolers: ICooler[];
	navigation: NavigationComponent;
}

export default class Coolers extends Component<IProps> {

	protected onPress = (id: ReactText) =>
		this.props.navigation.navigate('CoolerCard', {id})

	public render() {
		const {coolers} = this.props;

		return (
			<ScrollView style={styles.container}>
				{
					coolers.map((cooler: ICooler) => (
						<CoolerItem
							key={cooler.id}
							cooler={cooler}
							onPress={this.onPress}
						/>
					))
				}
			</ScrollView>
		);
	}
}
