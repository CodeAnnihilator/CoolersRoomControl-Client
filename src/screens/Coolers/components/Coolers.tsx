import React, {ReactText} from 'react';
import {StyleSheet, Text, View} from 'react-native';

import CoolerItem from './CoolerItem';
import CoolerSettings from './CoolerSettings';

const mock = [{
	id: 1,
	title: 'Cooler 1',
}, {
	id: 2,
	title: 'Cooler 2',
}, {
	id: 3,
	title: 'Cooler 3',
}];

const styles = StyleSheet.create({
	container: {
		flex: 1,
		alignItems: 'center',
		justifyContent: 'center',
	},
});

export default class Coolers extends React.Component<any, any> {
	public state = {
		selectedCoolerTitle: null,
	};

	private readonly getCoolerSettings = (selectedCoolerTitle: ReactText) =>
		this.setState({selectedCoolerTitle})

	protected onPress = (id: ReactText) =>
		this.props.navigation.navigate('CoolerCard', {id})

	public render() {

		const {selectedCoolerTitle} = this.state;
		const {navigation} = this.props;
		const roomID = JSON.stringify(navigation.getParam('roomID'));

		return (
			<View style={styles.container}>
				{
					selectedCoolerTitle && (
						<CoolerSettings selectedCoolerTitle={selectedCoolerTitle} />
					)
				}
				<Text>Coolers</Text>
				{
					roomID && (
						<Text>Room ID is {roomID}</Text>
					)
				}
				{
					mock.map(({id, title}) => (
						<CoolerItem
							key={id}
							id={id}
							title={title}
							onLongPress={this.getCoolerSettings}
							onPress={this.onPress}
						/>
					))
				}
			</View>
		);
	}
}
