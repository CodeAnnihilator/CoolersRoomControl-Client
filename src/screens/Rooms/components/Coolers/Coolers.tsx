import React, {ReactText, Component} from 'react';
import {NavigationComponent} from 'react-navigation';
import {ScrollView, Text} from 'react-native';

import {ICooler} from '../../types/Cooler';

import CoolerItem from '../../../../common/components/CoolerItem/CoolerItem';

interface IProps {
	coolers: ICooler[];
	navigation: NavigationComponent;
}

class Coolers extends Component<IProps> {

	protected onPress = (id: ReactText) =>
		this.props.navigation.navigate('CoolerCardByRoom', {id})

	public render() {
		const {coolers} = this.props;

		return (
			<ScrollView>
				{
					coolers.length ?
						coolers.map((cooler: ICooler) => (
							<CoolerItem
								key={cooler.id}
								cooler={cooler}
								onPress={this.onPress}
							/>
						)) :
						<Text>No coolers</Text>
				}
			</ScrollView>
		);
	}
}

export default Coolers;
