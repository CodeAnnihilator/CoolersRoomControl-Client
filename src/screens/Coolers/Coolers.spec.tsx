import * as React from 'react';
import renderer from 'react-test-renderer';

import Coolers from './Coolers';

describe('Coolers', () => {
	it('+++capturing Snapshot of Notifications', () => {
		const arrCoolers = [{
			id: 1,
			power: 'power',
			title: 'title',
			series: 'series',
			status: 'status',
			roomID: 1,
			place: 'place',
			source: 'source',
			type: 'type',
			certification: 'certification',
			cop: 'cop',
		}];
		const navigation = null;

		const renderedValue =  renderer.create(<Coolers coolers={arrCoolers} navigation={navigation} />).toJSON();

		expect(renderedValue).toMatchSnapshot();
});
});
