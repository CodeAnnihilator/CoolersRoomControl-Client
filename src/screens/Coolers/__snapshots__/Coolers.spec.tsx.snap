// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Coolers +++capturing Snapshot of Notifications 1`] = `null`;
